src/odemis/driver/test/ueye_test.py

```diff
@@ -97,5 +98,63 @@ def setUp(self):
         VirtualTestCam.setUp(self)
 
 
+class TestUEyeCroppingTranspose(VirtualTestCam, unittest.TestCase):
+    """
+    Test setting up the UEye class with a transposed image and a set resolution to force cropping.
+    """
+    camera_type = CLASS
+    camera_kwargs = KWARGS
+
+    @classmethod
+    def setUpClass(cls):
+        if TEST_NOHW:
+            raise unittest.SkipTest('No camera HW present. Skipping tests.')
+
+        # set a fixed max_res smaller than the maximum supported camera resolution
+        cls.camera_kwargs['max_res'] = (1000, 1000)
+        cls.camera_kwargs['transp'] = [2, -1]
+
+        super(TestUEyeCroppingTranspose, cls).setUpClass()
+
+    @classmethod
+    def tearDownClass(cls):
+        if TEST_NOHW:
+            return
+        super(TestUEyeCroppingTranspose, cls).tearDownClass()
+
+    def setUp(self):
+        if TEST_NOHW:
+            self.skipTest("No simulator available")
+
+        VirtualTestCam.setUp(self)
+
+    def test_set_resolution(self):
+        """
+        Test if the max resolution and the transpose argument are passed and applied.
+        Also test if an acquired image is of the correct dimensions and the axes are right also with binning active.
+        """
+        # assign the right resolutions from the initialization of the camera class
+        cam_resolution = self.camera._sensor_res
```

Comment:
I'm a bit confused by the names of these 3 resolutions. I'd suggest:
* cam_resolution -> sensor_res_raw
* trans_resolution -> expected_res  (and don't swap it!, just self.camera_kwargs['max_res'])
* set_resolution -> sensor_res_user